Index: agavi/util/AgaviToolkit.class.php
===================================================================
--- agavi/util/AgaviToolkit.class.php	(revision 100)
+++ agavi/util/AgaviToolkit.class.php	(working copy)
@@ -475,6 +475,15 @@
 	 */
 	public static function evaluateModuleDirective($moduleName, $directiveNameFragment, $variables = array())
 	{
+	    
+	    /*
+	     * Simple patch to create a new variable named shortActionName to make them available in expandVariables
+	     */
+	    if (isset($variables['actionName']))
+	    {
+            $variables['shortActionName'] = self::extractShortActionName($variables['actionName']);
+	    }
+	      
 		return AgaviToolkit::expandVariables(
 			AgaviToolkit::expandDirectives(
 				AgaviConfig::get(
@@ -487,8 +496,29 @@
 			),
 			$variables
 		);
-		
 	}
+	
+	/**
+     * Extracts the "shortActionName" from given action name . e.g.:
+     * 
+     * Foo/Bar -> Bar is now the shortActionName
+     *
+     * @param      string The name of the action
+     *
+     * @return     string The final value
+     *
+     * @author     Jan.Schütze <jan.schuetze@exozet.com>
+     * @since      1.0.2
+     */
+    public static function extractShortActionName($longActionName)
+    {   
+        $pos = strrpos($longActionName,'/');
+        if ($pos !== false) {
+            return substr($longActionName, $pos + 1);
+        }   
+
+        return $longActionName;
+    }
 }
 
 ?>
\ No newline at end of file
