Index: agavi/src/testing/AgaviTesting.class.php
===================================================================
--- agavi/src/testing/AgaviTesting.class.php	(revision 4918)
+++ agavi/src/testing/AgaviTesting.class.php	(working copy)
@@ -119,7 +119,6 @@
 				if(empty($suites[$name])) {
 					throw new InvalidArgumentException(sprintf('Invalid suite name %1$s.', $name));
 				}
-				
 				$master_suite->addTest(self::createSuite($name, $suites[$name]));		
 			}
 				
@@ -185,9 +184,32 @@
 	 */
 	protected static function createSuite($name, array $suite) 
 	{
+		$base = ! isset($suite['base']) ? 'tests' : $suite['base'];
+		if(!AgaviToolkit::isPathAbsolute($base)) {
+			$base = AgaviConfig::get('core.testing_dir').'/'.$base;
+		}
 		$s = new $suite['class']($name);
+		if(!empty($suite['includes'])) {
+			foreach(
+				new RecursiveIteratorIterator(
+					new AgaviRecursiveDirectoryFilterIterator(
+						new RecursiveDirectoryIterator($base),
+						$suite['includes'],
+						$suite['excludes']
+					),
+					RecursiveIteratorIterator::CHILD_FIRST
+				) as $finfo) {
+
+				if($finfo->isFile()) {
+					$s->addTestFile($finfo->getPathName());
+				}
+			}
+		}
 		foreach($suite['testfiles'] as $file) {
-			$s->addTestFile('tests/'.$file);
+			if(!AgaviToolkit::isPathAbsolute($file)) {
+				$file = $base.'/'.$file;
+			}
+			$s->addTestFile($file);
 		}
 		return $s;
 	}
